app/javascript/packages/verify-flow/steps/personal-key-confirm/personal-key-input.spec.tsx

```diff
@@ -49,12 +49,12 @@ describe('PersonalKeyInput', () => {
     expect(input.value).to.equal('1234-1234-1234-1234');
   });
 
-  it('validates the input value against the expected value', async () => {
-    const { getByRole } = render(<PersonalKeyInput expectedValue="0000-0000-0000-0000" />);
+  it('validates the input value against the expected value, case-insensitive', async () => {
+    const { getByRole } = render(<PersonalKeyInput expectedValue="abcd-0000-DEFG-0000" />);
 
     const input = getByRole('textbox') as HTMLInputElement;
 
-    await userEvent.type(input, '0000-0000-0000-000');
+    await userEvent.type(input, 'ABCD-0000-defg-000');
```

Comment:
Would it make sense to have a second test where the `values.toLowercase()` match and we expect not to see an error?

Reply:
Could you share a snippet of the test case you have in mind?

The idea with mixed case here and in the `expectedValue` above is to try to cover both upper- and lower- variations in either the expected value or the given value.

Reply:
oh one more thought! the keys are [base32-crockford encoded](http://www.crockford.com/base32.html), so that they fold `0` and `o` together, `1` and `i`, when we compare on the server we normalize them. We should probably do the same thing here while we're fixing

Reply:
link to source: https://github.com/18F/identity-idp/blob/db4c94a3ad95ac758eb31db21925c968f6d394c9/app/services/personal_key_generator.rb#L17-L38

Reply:
Good call @zachmargolis , I'll add that. I guess we don't have any feature specs covering that? (That's what originally prompted me to create this pull request)

I also forgot we had created [a package](https://github.com/18F/identity-idp/blob/main/app/javascript/packages/personal-key-input/index.js) for that purpose. Ideally we could fold that in here, but since we still have to maintain the old page, we might have to keep it around for a while in its current form.

Reply:
> I guess we don't have any feature specs covering that?

I'm not seeing any feature specs... but we do have unit test coverage of it:
https://github.com/18F/identity-idp/blob/db4c94a3ad95ac758eb31db21925c968f6d394c9/spec/services/personal_key_generator_spec.rb#L76-L78

Reply:
Enhanced feature spec coverage in [d4599a5](https://github.com/18F/identity-idp/pull/6240/commits/d4599a58a2dba596c39a6b4a42d25626f8c81014). Will implement it here shortly.

Reply:
Implemented in [a6404d6](https://github.com/18F/identity-idp/pull/6240/commits/a6404d6d1ddd06bed7a80cb7c6b6721728be148c). I could have pulled in the library we used in `@18f/identity-personal-key-input`, but it kinda seemed a bit overkill?